lib/spack/spack/config.py

```diff
@@ -1262,6 +1271,16 @@ def process_config_path(path):
             front = syaml.syaml_str(front)
             front.append = True
 
+        quote = "['\"]"
+        not_quote = "[^'\"]"
```

Comment:
(no change needed) will this allow `'inconsistentquotes"` or would YAML complain? I don't think it's necessary to enforce if YAML allows (since we get rid of them)

Reply:
I think we're divorcing from yaml conventions sufficiently that it doesn't matter from the user perspective and the quotes will be stripped before any actual yaml parsers see it, so while, yes it would allow for that, IMO it's a non issue.

Reply:
I'll mark this as resolved: it doesn't need to be altered.

Reply:
Sounds good to me. Thanks!